### PR TITLE
Add the crate that connects global allocator

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/5" target="_blank">Graphite</a> 
    * **PR #7** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/7" target="_blank">Graphite</a> 
      * **PR #6** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/6" target="_blank">Graphite</a> 
        * **PR #8** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/8" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/8?utm_source=stack-comment).

File: lang/allocator/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "sgx-std-lang-allocator"
+version = "0.1.0"
+edition = "2021"
+authors = ["MobileCoin"]
+rust-version = "1.62.1"
+license = "Apache-2.0"
+readme = "README.md"
+repository = "https://github.com/mobilecoinfoundation/sgx-std"
+description = "Allocator for SGX enclave"
```
Comment (nick-mobilecoin) [inline lang/allocator/Cargo.toml:10]: Up for suggestions on how to better differentiate this and the `mc-sgx-alloc` crate.  Currently they have the same description

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 [workspace]
 members = [
-    "alloc"
+    "alloc",
+]
+exclude = [
+    "lang/allocator",
```
Comment (nick-mobilecoin) [inline Cargo.toml:6]: @jcape Not sure if this was the folder hierarchy you were thinking of

**Review by awygle — APPROVED** [review]

File: lang/allocator/Cargo.toml (hunk 1 above)
Comment (jcape) [reply lang/allocator/Cargo.toml:10]: I think "Global Allocator Language Item for SGX Enclaves" is probably the way to go.

File: lang/allocator/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# MobileCoin SGX: Allocator
+
+[![Project Chat][chat-image]][chat-link]<!--
+-->![License][license-image]<!--
+-->![Target][target-image]<!--
+-->[![Crates.io][crate-image]][crate-link]<!--
+-->[![Docs Status][docs-image]][docs-link]<!--
+-->[![Dependency Status][deps-image]][deps-link]
+
+Provide allocator for use in an SGX enclave
```
Comment (jcape) [inline lang/allocator/README.md:10]: Note that this is the lang item, the allocator is provided by the `mc-sgx-alloc` crate.

**Review by jcape — APPROVED** [review]

File: lang/allocator/README.md (hunk 3 above)
Comment (nick-mobilecoin) [reply lang/allocator/README.md]: updated statement similar to the cargo toml description, also reworded header

**Comment by nick-mobilecoin** [comment]
Closing this as it was determined this could be done with a macro instead of an extra crate

**Pull request closed without merging**